### DISCUSSION TITLE
Can I use 2 pnp searchbox webparts and 2 pnp search result webparts on the same page?

### DISCUSSION THREAD

**Discussion by hajjkh** [microsoft-search/pnp-modern-search#3311] [Q&A]
I would like to add a "pnp searchbox webpart A" connected to "pnp search results webpart A" on a page. On the same page I would like to add "pnp searchbox webpart B" connected to "pnp search results webpart B" on the same page.

At this moment "pnp searchbox webpart A" give search results in both "pnp search results webpart A" and "pnp search results webpart B".

**Comment by kasperbolarsen** [comment]
Have you looked at using search verticals for targeting your search?

**Comment by kasperbolarsen** [comment]
On "page 3" of each result web part you can specify which searchbox web part the result web part will use.
